### DISCUSSION TITLE
if prisma loop inside transaction send queries one time and if it really efficient??

### DISCUSSION THREAD

**Discussion by Abd-elrahman2003** [prisma/prisma#24498] [Q&A] [answered]
### Question

hello guys i need to ask question about prisma
i asked GPT about prisma transaction and he told me that it is establish one connection
is that really 
second if i have perform loop inside trasnaction like

prisma.$transaction(async prisma=>{
     items.map((val, index)=>{
        prisma.create({data:val})
     })
})
if that make all queries on DB server all one time with out send a lot of connection request or send each query one by one 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by janpio** [comment]
A database transaction limits all queries sent inside it to the same database connection, as a transaction can only live on one connection. So yes, all queries executed inside it are also executed sequentially, as one database connection can only execute one query at a time.

**Reply by Abd-elrahman2003** [reply]
ok nice thank you but i have a small question 

if sequentially execution is bad for app performance or it is normal if just one connection has established

in other way if i loop inside transaction for 100 iteration and each loop make query is that send to db server 100 request and if happend; is that a big case?

**Reply by janpio** [reply, marked as answer]
> if sequentially execution is bad for app performance or it is normal if just one connection has established

Inside a transaction, that is pretty much the only way. Apps usually read more data than they write, where not transactions are required, and there it makes total sense to send queries in parallel instead of sequentually.

> in other way if i loop inside transaction for 100 iteration and each loop make query is that send to db server 100 request and if happend; is that a big case?

If you are sending 100 queries in a loop, _that_ is the problem. Find a way to combine these queries by for example collecting all the data in the loop first, and then sending one query instead.

**Reply by Abd-elrahman2003** [reply]
ok thank you very much, that was very useful
